digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/MunicipalityApi.scala

```diff
@@ -44,1089 +37,56 @@ class MunicipalityApi(val onOffLinearAssetService: OnOffLinearAssetService,
-    val usedService = verifyLinearServiceToUse(assetTypeId)
-    val newLinearAssets = extractNewLinearAssets(assetTypeId, parsedBody)
-    validateSideCodes(newLinearAssets)
-    newLinearAssets.foreach{
-      newAsset => validateMeasuresOnAssets(Set(newAsset.startMeasure, newAsset.endMeasure), newAsset.linkId)
-    }
-    val assetsIds = usedService.create(newLinearAssets, assetTypeId, user.username)
-    getLinearAssetsAndRoadLinks(assetTypeId, assetsIds.toSet)
-  }
-
-  def createSpeedLimitAssets(assetTypeId: Int, parsedBody: JValue): Seq[(SpeedLimit, RoadLink)] ={
-    val newSpeedLimitAssets = extractNewLinearAssets(assetTypeId, parsedBody)
-    validateSideCodes(newSpeedLimitAssets)
-    newSpeedLimitAssets.foreach{
-      newAsset => validateMeasuresOnAssets(Set(newAsset.startMeasure, newAsset.endMeasure), newAsset.linkId)
-    }
-    val assetsIds = speedLimitService.createMultiple(newSpeedLimitAssets, assetTypeId, user.username, 0, (_, _) => Unit)
-    getSpeedLimitsAndRoadLinks(assetsIds.toSet)
-  }
-
-  private def getLinkIds(manoeuvres: Seq[NewManoeuvreValues]): Seq[Long] = {
-    manoeuvres.map(_.properties.find(_.name == "sourceLinkId").map(_.value.asInstanceOf[BigInt].toLong).getOrElse(halt(NotFound("sourceLinkId not found"))))++
-      manoeuvres.flatMap(_.properties.find(_.name == "elements").map(_.value.asInstanceOf[Seq[BigInt]].map(_.toLong)).getOrElse(Seq())) ++
-      manoeuvres.map(_.properties.find(_.name == "destLinkId").map(_.value.asInstanceOf[BigInt].toLong).getOrElse(halt(NotFound("destLinkId not found"))))
-  }
+  put("/assetUpdateFromAWS") {
+    try {
+      val jsonDatasets: List[List[Any]] = parsedBody.extractOrElse[List[List[Any]]](throw new ClassCastException)
 
-  def createManoeuvreAssets(parsedBody: JValue): Seq[(Manoeuvre, Seq[RoadLink])] = {
-    val manoeuvreIds = parsedBody.extractOpt[Seq[NewManoeuvreValues]].map { manoeuvres =>
+      val listDatasets = jsonDatasets.map(data =>
+        Dataset(data(0).asInstanceOf[String], data(1).asInstanceOf[Map[String, Any]], data(2).asInstanceOf[List[List[BigInt]]])
+      )
 
-      val roadLinks = roadLinkService.getRoadsLinksFromVVH(getLinkIds(manoeuvres).toSet)
-      validateManoeuvrePropForCreate(manoeuvres, roadLinks)
+      var datasetFeaturesWithoutIds = Map[String, Int]()
 
-      manoeuvres.map {manoeuvre =>
-        val linkIds = getLinkIds(Seq(manoeuvre))
-        manoeuvreService.createManoeuvre(user.username,
-          NewManoeuvre(
-            convertValidityPeriod(manoeuvre.properties.find(_.name == "validityPeriods")).getOrElse(Seq()).toSet,
-            manoeuvre.properties.find(_.name == "exceptions").map(_.value.asInstanceOf[List[BigInt]].map(_.toInt)).getOrElse(Seq()),
-            manoeuvre.properties.find(_.name == "additionalInfo").map(_.value.toString), linkIds, None),
-         roadLinks.filter(road => linkIds.contains(road.linkId))
+      OracleDatabase.withDynTransaction {
```

Comment:
In other APIs in the project we not controlling the transactions on the API, but in the service class. We could do the same here